pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -43,14 +43,22 @@ let
   );
 
   forceRust = features.rust or false;
-  kernelSupportsRust = lib.versionAtLeast version "6.7";
+  kernelSupportsRust = (
+    lib.versionAtLeast version "6.12"
+    && (
+      stdenv.hostPlatform.isx86_64
+      || stdenv.hostPlatform.isLoongArch64
+      || stdenv.hostPlatform.isAarch64
+      || (stdenv.hostPlatform.isRiscV64 && !stdenv.cc.isGNU)
+    )
+  );
 
   # Currently only enabling Rust by default on kernel 6.12+,
   # which actually has features that use Rust that we want.
   defaultRust = lib.versionAtLeast version "6.12" && rustAvailable;
```

Comment:
We can drop the version check here too, and maybe this entire variable? Or maybe this should be `kernelSupportsRust && rustAvailable`? But then it won't be a default, it'll just be a `canWeActuallyBuildWithRust` or something. Any suggestions on cleaning this up?

Reply:
I think I'll just drop the entire variable, but as a separate commit, in case someone wants to come back and extend the compatibility list for pre-6.12 again.

Reply:
Thought about this while trying to implement it, wouldn't it make sense to have the "force Rust" flag affect the version check (i.e. `kernelSupportsRust`) instead of the `rustAvailable` flag? We are admittedly very lax with the exact version steps, so it would be more beneficial to the user to override that decision rather than "you don't have `rustc` installed".

Reply:
Hm, I thought we already did that?

Reply:
Nope, the current logic is `(forceRust || defaultRust) && kernelSupportsRust;` (and with `kernelSupportsRust` only starting from 6.12 now, that is now a hard requirement for enabling rust and not forceable).

Reply:
Done, and I tried making the surrounding bits a bit nicer as well.